File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        for num in nums:
            count[num] = count.get(num, 0) + 1
        
        arr = []
        for val, cnt in count.items():
            arr.append([cnt, val])
        arr.sort()

        res = []
        while len(res) < k:
            res.append(arr.pop()[1])
        return res

File: Data Structures & Algorithms/three-integer-sum/submission-1.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> res = new ArrayList<>();
        int n = nums.length;

        for (int i=0; i<n-2; i++) {
            int l = i+1;
            int r = n-1;
            if (i>0 && nums[i]==nums[i-1]) {
                continue;
            }

            while (l<r) {
                int threeSum = nums[i] + nums[l] + nums[r];
                if (threeSum > 0) {
                    r--;
                }else if (threeSum < 0) {
                    l++;
                }else {
                    List<Integer> threeSumList = Arrays.asList(nums[i], nums[l], nums[r]);
                    l++;
                    r--;
                    res.add(threeSumList);
                    while (l<r && nums[l] == nums[l-1]) {
                        l++;
                    }
                    
                }
            }
        }
        return res;
    }
}
